fastparquet/test/test_writer.py

```diff
@@ -223,6 +222,48 @@ def test_nulls_roundtrip(tempdir):
         assert (data[col].isnull() == df[col].isnull()).all()
 
 
+def test_make_definitions_with_nulls():
+    for _ in range(10):
```

Comment:
There is a different length each time, and whether we have an exact multiple of 8 or not is important.